src/raw/alloc.rs

```diff
@@ -0,0 +1,29 @@
+pub use self::inner::*;
+
+#[cfg(feature = "nightly")]
+mod inner {
+    pub use crate::alloc::alloc::{Alloc, Global};
+}
+
+#[cfg(not(feature = "nightly"))]
+mod inner {
+    use core::ptr::NonNull;
+    use crate::alloc::alloc::{Layout, alloc, dealloc};
+
+    pub trait Alloc {
+        unsafe fn alloc(&mut self, layout: Layout) -> Result<NonNull<u8>, ()>;
+        unsafe fn dealloc(&mut self, ptr: NonNull<u8>, layout: Layout);
+    }
+
+    #[derive(Copy, Clone)]
+    pub struct Global;
+    impl Alloc for Global {
+        unsafe fn alloc(&mut self, layout: Layout) -> Result<NonNull<u8>, ()> {
+            Ok(NonNull::new_unchecked(alloc(layout)))
```

Comment:
This is wrong: global allocation can fail, in which case it returns a null pointer. You should use `NonNull::new` which returns a `Result`.

Reply:
Thanks! The documentation didn't that behavior so I assumed it would abort.